api/graphql/reservations/reservation_serializers/confirm_serializers.py

```diff
@@ -79,15 +79,20 @@ def validate(self, data):
             payment_type = data.get("payment_type", "").upper()
             reservation_unit = self.instance.reservation_unit.first()
 
-            if not reservation_unit.payment_product:
+            if self.instance.price_net > 0 and not reservation_unit.payment_product:

```

Comment:
what if reservation_unit `pricing_type` is `FREE`? In that case I think the price validations should be ignored.

Reply:
Krista confirmed this and said that it is not possible to have `price > 0` if `pricing_type` is `FREE`. We already have validation in place in the UI and also price calculations are skipped for `FREE` reservation units. So I'd say we don't need extra validation for this because it basically means data has been tweaked directly via database or via Django Admin.